app/controllers/likes_controller.rb

```diff
@@ -57,13 +62,28 @@ def destroy
   end
 
   private
-    # Use callbacks to share common setup or constraints between actions.
-    def set_like
-      @like = Like.find(params[:id])
-    end
 
-    # Only allow a list of trusted parameters through.
-    def like_params
-      params.require(:like).permit(:fan_id, :photo_id)
-    end
+  def set_photo
+    @photo = Photo.find(params[:like][:photo_id])
+  end
+
+  def set_like
+    @like = Like.find(params[:id])
+  end
+
+  def authorize_resource
+    authorize @like || Like
+  end
+
+  def authorize_create_action
+    authorize @photo, :create_like?
+  end
+
+  def record_not_found
+    redirect_to likes_path, alert: 'Like not found.'
+  end
+
+  def user_not_authorized
+    redirect_to(request.referrer || root_path, alert: 'You are not authorized to perform this action.')
+  end
```

Comment:
I like the idea of authorizing the whole resource instead of individually authorizing the actions. Very clever!